commit caa8290510244990b26106e027aa253237ada629
Author: Nicholas Marriott <nicm@openbsd.org>
Date:   Fri Apr 12 12:44:31 2013 +0000

    Copy the client into the new cmdq in source-file so commands that work
    on it (such as new-session) can work. Fixes issue reported by oss-adv at
    users dot sf dot net.

diff --git a/cmd-source-file.c b/cmd-source-file.c
index 1bd2bb0..f50efbe 100644
--- a/cmd-source-file.c
+++ b/cmd-source-file.c
@@ -49,6 +49,7 @@ cmd_source_file_exec(struct cmd *self, struct cmd_q *cmdq)
 	char		*cause;
 
 	cmdq1 = cmdq_new(NULL);
+	cmdq1->client = cmdq->client;
 	cmdq1->emptyfn = cmd_source_file_done;
 	cmdq1->data = cmdq;
 
